modules/openapi-generator/src/main/java/org/openapitools/codegen/languages/SpringCodegen.java

```diff
@@ -733,10 +733,7 @@ public void preprocessOpenAPI(OpenAPI openAPI) {
                                 value.put("tag", tag);
                                 tags.add(value);
                             }
-                            if (operation.getTags().size() > 0) {
-                                final String tag = operation.getTags().get(0);
-                                operation.setTags(Arrays.asList(tag));
-                            }
+
```

Comment:
Won't this now _not_ use tags at all since you removed `operation.setTags()`?

Reply:
No because tags are already used, we are just overriding those tags with first tag only.